h2o-extensions/target-encoder/src/main/java/ai/h2o/targetencoding/TargetEncoderModel.java

```diff
@@ -147,38 +150,47 @@ public TargetEncoderModel(Key<TargetEncoderModel> selfKey, TargetEncoderParamete
   }
 
   public Frame transformTraining(Frame fr) {
-    return transform(fr, _parms.getBlendingParameters(), _parms._noise, true);
+    return transformTraining(fr, NO_FOLD);
+  }
+  
+  public Frame transformTraining(Frame fr, int outOfFold) {
+    assert outOfFold == NO_FOLD || _parms._data_leakage_handling == DataLeakageHandlingStrategy.KFold;
+    return transform(fr, true, outOfFold, _parms.getBlendingParameters(), _parms._noise);
```

Comment:
this method is used when applying TE during CV.
This ensures that there's no leakage from out of fold data into the cv training data.